contactql/visitor.go

```diff
@@ -75,31 +75,43 @@ func (v *visitor) VisitImplicitCondition(ctx *gen.ImplicitConditionContext) inte
 	if v.redaction == envs.RedactionPolicyURNs {
 		num, err := strconv.Atoi(value)
 		if err == nil {
-			return newCondition(PropertyTypeAttribute, AttributeID, "=", strconv.Itoa(num), attributes[AttributeID])
+			return newCondition(PropertyTypeAttribute, AttributeID, ComparatorEqual, strconv.Itoa(num), attributes[AttributeID])
 		}
 	} else if asURN != urns.NilURN {
 		scheme, path, _, _ := asURN.ToParts()
 
-		return newCondition(PropertyTypeScheme, scheme, "=", path, assets.FieldTypeText)
+		return newCondition(PropertyTypeScheme, scheme, ComparatorEqual, path, assets.FieldTypeText)
 
 	} else if telRegex.MatchString(value) {
 		value = cleanSpecialCharsRegex.ReplaceAllString(value, "")
 
-		return newCondition(PropertyTypeScheme, urns.TelScheme, "~", value, assets.FieldTypeText)
+		return newCondition(PropertyTypeScheme, urns.TelScheme, ComparatorContains, value, assets.FieldTypeText)
 	}
 
-	return newCondition(PropertyTypeAttribute, AttributeName, "~", value, attributes[AttributeName])
+	// convert to contains condition only if we have the right tokens, otherwise make equals check
+	comparator := ComparatorContains
+	if len(tokenizeNameValue(value)) == 0 {
+		comparator = ComparatorEqual
+	}
```

Comment:
So effectively if user types `X`, instead of erroring that `X` is too short to make `name ~ "X"`, we rewrite query as `name = "X"`